Api/Validator.php

```diff
@@ -0,0 +1,220 @@
+<?php
+    
+    namespace Metrilo\Analytics\Api;
+    
+    class Validator {
+        public $logDestination = __DIR__ . '/MetriloApiErrors.log'; // construct log path in ajax controller and pass it as constructor param for Validator class
+    
+        public $errors = [];
+    
+        public function check($var){
+            $this->var = $var;
+            
+            return $this;
+        }
+    
+        public function value($value){
+            $this->value = $value;
+            
+            return $this;
+        }
+    
+        public function required(){
+            if($this->value == '' || $this->value == null || $this->value == []){
+                $this->errors[] = 'Field ' . $this->var . ' is required. ';
+            }
+            
+            return $this;
+        }
+        
+        public function isSuccess(){
+            if(empty($this->errors)) return true;
+        }
+    
+        public function getErrors(){
+            if(!$this->isSuccess()) return $this->errors;
+        }
+    
+        public function isString(){
+            if(!is_string($this->value)) {
+                $this->errors[] = 'Field ' . $this->var . ' is not String type. ';
+            }
+            
+            return $this;
+        }
+    
+        public function isInt(){
+            if(!is_int($this->value)) {
+                $this->errors[] = 'Field ' . $this->var . ' is not Integer type. ';
+            }
+            
+            return $this;
+        }
+    
+        public function isEmail(){
+            if(!filter_var($this->value, FILTER_VALIDATE_EMAIL)) {
+                $this->errors[] = 'Field ' . $this->var . ' is not a valid email address. ';
+            }
+            
+            return $this;
+        }
+    
+        public function isNumeric(){
+            if(!is_numeric($this->value)) {
+                $this->errors[] = 'Field ' . $this->var . ' is not Number. ';
+            }
+            
+            return $this;
+        }
+        
+        public function logger($error) {
+            $validationErrors = implode("| ", $this->getErrors());
+            $error           .= $validationErrors;
+            return error_log($error . PHP_EOL, 3, $this->logDestination);
+        }
+        
+        public function validateCallParams($callParams = []) {
```

Comment:
What is this? It is not used

Reply:
Removed param validation but forget to remove the validation method. Removed.